feat: global default parameters

frederickfogerty opened the pull request:
## Description

This PR allows developers to set default parameters for every img tag. This prevents them from having to set the same parameters over and over again.

This fixes #129 

## Steps to test

Review new unit tests.

jayeb commented on src/ImgixTag.js:
```diff
@@ -49,10 +49,14 @@ var ImgixTag = (function() {
-    var params;
+		var params = {};
+
+		if (this.settings.defaultParams) {
```
Should we type-check this to make sure it's an object?

frederickfogerty replied:
Yep, I can do this

jayeb commented on src/ImgixTag.js:
```diff
@@ -49,10 +49,14 @@ var ImgixTag = (function() {
 
   ImgixTag.prototype._extractBaseParams = function() {
-    var params;
+		var params = {};
```
[nit] Spaces and tabs, my friend... spaces and tabs.

frederickfogerty replied:
🙈 https://www.youtube.com/watch?v=cowtgmZuai0

frederickfogerty commented:
1. I've added two tests for this in `spec/ImgixTagSpec.js` - are these sufficient? 

2. I think this is a good idea - I'll update this PR once I've implemented this.

frederickfogerty commented:
@jayeb (1) was discussed offline, and I believe I've implemented (2). Could you give this another review?

jayeb commented on src/ImgixTag.js:
```diff
@@ -81,7 +83,7 @@ var ImgixTag = (function() {
     if (this.settings.includeLibraryParam) {
       params.ixlib = 'imgixjs-' + imgix.VERSION;
-    }
+		}
```
[nit] Tabs & spaces, again. Is this perchance a Prettify thing that's breaking down?

frederickfogerty replied:
Gah! I think it's time to tame the beast and work out what's going on here.

jayeb commented on src/defaultConfig.js:
```diff
@@ -2,7 +2,8 @@ module.exports = {
   host: null,
   useHttps: true,
-  includeLibraryParam: true,
+	includeLibraryParam: true,
```
[nit] T&S

frederickfogerty commented:
@jayeb those pesky space and tab issues should be gone for good now!
